### PR TITLE
Hopper

### PR DESCRIPTION
**Pull request by ManiacMoustache**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] I have read the **CONTRIBUTING** document.
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

## Associated Issue(s)
Issue: #60 
## Steps to Test
1. Get teapot
2. Test automatic/manual loading
3. See if balls show up in networktables
4. If everything works, good.

## Changelog
Beefed up ballcounter, hoppermotor. Cut down loaderLogic

## Why?
So that we can use the hopper on teapot with a large amount of information gathered.

### CODE REVIEW

File: components/Input/ballCounter.py [hunk 1]
```diff
@@ -1,69 +1,138 @@
...
+    compatString = ["doof", "teapot"]
     sensors: Sensors
+    colorSensor: ColorSensor
     shooter: ShooterLogic
-    maxBalls = 4
+
+    def setup(self):
+        self.SmartTable = NetworkTables.getTable("SmartDashboard")
+        self.hopperTable = NetworkTables.getTable("components").getSubTable("hopperMotor")
+        readableArr = ["None", "None"]
```
Comment (tuffinmuffin) [inline components/Input/ballCounter.py:41]: Is this supposed to be None type:

File: components/Actuators/HighLevel/hopperMotor.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        Only changes motor speeds if they are 0
+        """
+        if self.hopperSpeed1 == 0 and self.sensors.loadingSensor(State.kTripped):
+            self.runHopperMotor1(self.intakeSpeed, Direction.kForwards)
+        if self.hopperSpeed2 == 0 and self.sensors.middleSensor(State.kTripped):
+            self.runHopperMotor2(self.movingSpeed, Direction.kForwards)
+
+    def execute(self):
+        if self.hopper1:
+            self.hopperMotor1.set(self.hopperSpeed1)
```
Comment (tuffinmuffin) [inline components/Actuators/HighLevel/hopperMotor.py:111]: hopperMotor1 and 2 are a bit confusing. At very least maybe add some docs to their defs or maybe lower / upper / front / back

**Review by Leaf-life — APPROVED** [review]
GOOD

File: components/Actuators/HighLevel/hopperMotor.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    motors_hopper: dict
+    sensors: Sensors
+    intakeSpeed = tunable(.1)
+    movingSpeed = tunable(.08)
+
+    def setup(self):
+        """
+        Sets up hopper motors
+        """
+        self.hopperSpeedFore = 0
```
Comment (tuffinmuffin) [inline components/Actuators/HighLevel/hopperMotor.py:22]: Future change and use just hopperSpeedFore == 0.0 and remove the bool, same for Back.

Comment (tuffinmuffin) [reply components/Actuators/HighLevel/hopperMotor.py:22]: and places checking for it running can call get the getter that knows 0 means not running

**Review by tuffinmuffin — APPROVED** [review]

**Review by hell0w0rld2 — APPROVED** [review]
👍👍👍

**Pull request merged**